### PR TITLE
cfunction => @cfunction

### PR DESCRIPTION
**Pull request by dmbates**
Quiet the warning about using `@cfunction`

### CODE REVIEW

File: src/Rmath.jl [hunk 1]
```diff
@@ -42,14 +42,26 @@ end
...
+        unsafe_store!(cglobal((:norm_rand_ptr,libRmath),Ptr{Cvoid}),
+                      cfunction(randn,Float64,Tuple{}))
+        unsafe_store!(cglobal((:exp_rand_ptr,libRmath),Ptr{Cvoid}),
+                      cfunction(Random.randexp,Float64,Tuple{}))
+    end
+else
+    function __init__()
+        # initialize RNG hooks
+        unsafe_store!(cglobal((:unif_rand_ptr,libRmath),Ptr{Cvoid}),
+                      @cfunction(rand,Float64,()))
```
Comment (ararslan) [inline src/Rmath.jl]: The macro is being expanded too early which is causing failures on 0.6, so what I think you could do is reverse the conditional and put this stuff first in the first block with a `@static` on the condition.

Comment (dmbates) [reply src/Rmath.jl]: I understand reversing the conditional.  Can you expand a bit on where the `@static` would go?

Comment (ararslan) [reply src/Rmath.jl]:
```julia
@static if VERSION >= v"0.7.0-DEV.4749"
    # stuff with @cfunction
else
    # stuff with cfunction
end
```

**Comment by dmbates** [comment]
Closes #51

**Pull request merged**